### DISCUSSION TITLE
Add ContinuousIntegrationBuild to DotNetCorePackSettings

### DISCUSSION THREAD

**Discussion by RehanSaeed** [cake-build/cake#3437] [Ideas]
`ContinuousIntegrationBuild` is Documented in https://devblogs.microsoft.com/dotnet/producing-packages-with-source-link/#deterministic-builds and is important for creating deterministic builds on different machines and operating systems. Suggested API:

```cs
new DotNetCorePackSettings()
{
    ContinuousIntegrationBuild = true
};
```

**Comment by augustoproiete** [comment]
LGTM

Current workaround is to use `WithProperty`:

```cakescript
DotNetCoreBuild("./SolutionName.sln", new DotNetCoreBuildSettings
{
    // ...

    MSBuildSettings = new DotNetCoreMSBuildSettings()
            .WithProperty("ContinuousIntegrationBuild", BuildSystem.IsLocalBuild ? "false" : "true")
});
```

**Reply by RehanSaeed** [reply]
Can we turn this into an issue?

**Reply by devlead** [reply]
Done
https://github.com/cake-build/cake/issues/3447

**Reply by augustoproiete** [reply]
@RehanSaeed I noticed you suggested adding this argument to `DotNetCorePackSettings` and this parameter is an MSBuild property that needs to be present during build (not pack) e.g. via `DotNetCoreBuild` unless, of course, when `DotNetCorePack` causes a build to be triggered first, so in both cases, it seems `DotNetCoreMSBuildSettings` seems to be the appropriate place for adding this property.

Thoughts?

**Reply by RehanSaeed** [reply]
My understanding is that it changes file paths in the nupkg, so you need to enable it in pack. I'm not sure about build. Happy to find a definitive answer.

But yes your suggestion makes total sense.

**Reply by augustoproiete** [reply]
@RehanSaeed PR open https://github.com/cake-build/cake/pull/3460
